core/network-libp2p/src/custom_proto/behaviour.rs

```diff
@@ -499,12 +499,14 @@ where
 	fn inject_dial_failure(&mut self, peer_id: Option<&PeerId>, addr: &Multiaddr, error: &dyn error::Error) {
 		if let Some(peer_id) = peer_id.as_ref() {
 			debug!(target: "sub-libp2p", "Failed to reach peer {:?} through {} => {:?}", peer_id, addr, error);
-			if self.connected_peers.contains(peer_id) {
-				self.topology.set_unreachable(addr);
-			}
+			self.topology.set_unreachable(addr);
 
 			// Trigger a `connect_to_nodes` round.
 			self.next_connect_to_nodes = Delay::new(Instant::now());
```

Comment:
Why is not triggered in the `else` case?

Reply:
The `else` only happens if we dial an address without knowing the `PeerId` that we expect to found, which only happens in one specific case: if the user passes a bootstrap node with the format `IP:PORT`.

Reply:
If it's a non-obvious implication of a logical assertion, then perhaps a comment would serve to inform future generations of maintainers? :-)